### PR TITLE
insert vuln api flag

### PR DESCRIPTION
**Pull request by ayushaga14**

### CODE REVIEW

File: apps/testing/src/main/java/com/akto/test_editor/strategy/InsertDataUtil.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import com.akto.parsers.HttpCallParser;
+import com.akto.utils.AccountHTTPCallParserAktoPolicyInfo;
+import com.akto.utils.DataInsertionUtil;
+
+public class InsertDataUtil {
+    
+    public static Map<Integer, AccountHTTPCallParserAktoPolicyInfo> accountHTTPParserMap = new ConcurrentHashMap<>();
+
+
+    public static void insertDataInApiCollection(int apiCollectionId, String topic, List<String> messages, List<String> errors, boolean skipKafka) throws Exception {
```
Comment (notshivansh) [inline apps/testing/src/main/java/com/akto/test_editor/strategy/InsertDataUtil.java:22]: skipKafka is an unused variable in this context.

Comment (notshivansh) [inline apps/testing/src/main/java/com/akto/test_editor/strategy/InsertDataUtil.java:22]: skipKafka is an unused variable in this context.

File: apps/testing/src/main/java/com/akto/test_editor/strategy/Strategy.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+import com.akto.dto.testing.YamlTestResult;
+import com.akto.log.LoggerMaker;
+import com.akto.log.LoggerMaker.LogDb;
+import com.akto.test_editor.Utils;
+
+public class Strategy {
+    
+    private static final String AKTO_DISCOVERED_APIS_COLLECTION = "shadow_apis";
+    private static final int AKTO_DISCOVERED_APIS_COLLECTION_ID = 1333333333;
+    private static final LoggerMaker loggerMaker = new LoggerMaker(Strategy.class);
```
Comment (notshivansh) [inline apps/testing/src/main/java/com/akto/test_editor/strategy/Strategy.java:21]: use this prototype instead, https://github.com/akto-api-security/akto/blob/master/libs/utils/src/main/java/com/akto/log/LoggerMaker.java#L73 the one being used is depracted.

File: apps/testing/src/main/java/com/akto/test_editor/strategy/Strategy.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+                if (!testRes.getVulnerable()) {
+                    continue;
+                }
+                if (testRes instanceof TestResult) {
+                    TestResult res = (TestResult) testRes;
+                    String message = res.getMessage();
+                    try {
+                        harPayload = Utils.convertToHarPayload(message, Context.accountId.get(), Context.now(), "", "HAR");
+                    } catch (Exception e) {
+                        loggerMaker.errorAndAddToDb("Error while converting attempt to har payload " + e.getMessage(), LogDb.TESTING);
```
Comment (notshivansh) [inline apps/testing/src/main/java/com/akto/test_editor/strategy/Strategy.java:42]: We can remove the second param, after changing the function call of the loggermaker fn.

File: apps/testing/src/main/java/com/akto/test_editor/Utils.java [hunk 4]
```diff
@@ -576,4 +577,44 @@ public static boolean evaluateResult(String operation, boolean currentRes, boole
...
+        }
+
+        String requestHeaders = mapper.writeValueAsString(reqHeaders);
+        String responseHeaders = (String) resp.get("headers");
+
+        String path = OriginalHttpRequest.getFullUrlWithParams((String) req.get("url"), (String) req.get("queryParams"));
+        String contentType = (String) reqHeaders.get("content-type");
+
+        Map<String,String> result = new HashMap<>();
+        result.put("akto_account_id", akto_account_id+"");
```
Comment (notshivansh) [inline apps/testing/src/main/java/com/akto/test_editor/Utils.java:602]: Use this instead https://github.com/akto-api-security/akto/blob/master/libs/utils/src/main/java/com/akto/open_api/parser/Parser.java#L309

File: apps/api-runtime/src/main/java/com/akto/utils/DataInsertionUtil.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+import com.akto.dao.AccountSettingsDao;
+import com.akto.dto.AccountSettings;
+import com.akto.dto.HttpResponseParams;
+import com.akto.parsers.HttpCallParser;
+import com.akto.runtime.APICatalogSync;
+
+public class DataInsertionUtil {
+
+    public static void processTraffic(HttpCallParser httpCallParser, List<HttpResponseParams> responses, int apiCollectionId){
```
Comment (notshivansh) [inline apps/api-runtime/src/main/java/com/akto/utils/DataInsertionUtil.java:13]: Use the same function in https://github.com/akto-api-security/akto/blob/master/apps/dashboard/src/main/java/com/akto/utils/Utils.java#L420

File: apps/testing/src/main/java/com/akto/test_editor/Utils.java (hunk 4 above)
Comment (ayushaga14) [reply apps/testing/src/main/java/com/akto/test_editor/Utils.java:602]: the constants are not properly defined. Let's discuss this later and modify, for now ignoring this comment

File: apps/api-runtime/src/main/java/com/akto/utils/DataInsertionUtil.java (hunk 5 above)
Comment (ayushaga14) [reply apps/api-runtime/src/main/java/com/akto/utils/DataInsertionUtil.java:13]: skippng this now, will modify this as a part of different pr

**Review by Ark2307 — APPROVED** [review]

**Pull request merged**